### ISSUE TITLE
Unexpected "statements are not aligned" error for unnecessary semicolon

### ISSUE THREAD

**Issue by sbj42** [palantir/tslint#2608]
### Bug Report

The `align` rule is reporting "statements are not aligned" for unnecessary semicolons.  This isn't a huge problem, because one should not use unnecessary semicolons anyway, but it is confusing and it would be better if the align rule left that up to the `semicolons` rule, whose complaint "Unnecessary semicolon" is more to the point. 

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
function test() {
};
```

with `tslint.json` configuration:

```json
{
  "rules": {
      "align": [true, "statements"]
  }
}
```

#### Actual behavior

```
ERROR: test.ts[2, 2]: statements are not aligned
```

#### Expected behavior

No error.

**Comment by sbj42** [comment]
I looked into this.  I think this behavior began at commit c1a167e, where [the align statements test was updated](https://github.com/palantir/tslint/commit/c1a167efdd7348ff95432f9e182b46cdaedae5a3#diff-0c32fd63c8f98d076dc22cd354549981) to make sure this behavior happens.  Looks like this was either part of fixing #2366, or of a general enhancement to the align rule.

I think this particular behavior is incorrect.  Yes, technically the empty statement created by the semicolon is a statement, and yes it doesn't align with the previous statement.  But I think this results in a confusing message to the user.

I suspect it can be easily avoided by ignoring empty statements in this rule.  For instance, by changing `node.statements` at `alignRule.ts:85` to `node.statements.filter(s => !isEmptyStatement(s))`.

**Comment by ajafff** [comment]
> I suspect it can be easily avoided by ignoring empty statements in this rule. For instance, by changing node.statements at alignRule.ts:85 to node.statements.filter(s => !isEmptyStatement(s)).

@sbj42 You're right, that's the only thing needed to fix this. Would you mind to submit a PR with that fix?

Unnecessary semicolons in classes will result in a similar problem with option `"members"`. There should be a filter for `SemicolonClassElement` added to line 138.

**Comment by sbj42** [comment]
Done.  So far I can't seem to get `"members"` to work.  I might be just doing it wrong.

**Comment by sbj42** [comment]
I'm satisfied with the resolution of this issue.  @ajafff mentioned a possible related issue at [this comment](https://github.com/palantir/tslint/issues/2608#issuecomment-298176322), but I don't know how to cause it - perhaps that can be submitted as a separate issue if someone produces a test case for it.

**Comment by ajafff** [comment]
@sbj42 I just opened #2668 to fix the problem with SemicolonClassElement that I mentioned above
